modules/gdscript/gdscript_compiler.cpp

```diff
@@ -2265,13 +2265,15 @@ Error GDScriptCompiler::_populate_class_members(GDScript *p_script, const GDScri
 
 	GDScriptDataType base_type = _gdtype_from_datatype(p_class->base_type, p_script);
 
+	int native_idx = GDScriptLanguage::get_singleton()->get_global_map()[base_type.native_type];
+	p_script->native = GDScriptLanguage::get_singleton()->get_global_array()[native_idx];
```

Comment:
Just to be sure, there's no risk of out of bound reads here?

It's the same code as there was for `NATIVE` but since it will now run for more base types, I think it might be worth double checking.

Reply:
Should not happen but I guess it doesn't hurt to add an extra check.